### ISSUE TITLE
Can't save promptValues without disabling skipLocalCache 

### ISSUE THREAD

**Issue by LTegtmeier** [yeoman/environment#298]
## Description

When using `yo@^4`, the `store: true` setting on prompts no longer saves the answer in the local `.yo-rc.json` like it did with `yo@^3`.

As shown below, that is because `skipLocalCache` is defaulted to `true`. Was this change intentional? Should it only disable the local cache in experimental mode?
https://github.com/yeoman/environment/blob/550757187894de6038546e1d93e2a6ac2be51135/lib/environment.js#L344-L349

#### Generator Code:

``` js
const Generator = require('yeoman-generator');

module.exports = class extends Generator {
    async prompting() {
        await super.prompt([{
            name: 'projectName',
            message: 'What is the name of your project?',
            store: true
        }]);
    }

    async writing() {
        this.config.set('some config', 'some value')
    }
};

```

#### Yeoman 3 .yo-rc.json:

`npm i -g yo@3.1.1 && yo ./generators/app`
``` json
{
  "generator-name": {
    "some config": "some value",
    "promptValues": {
      "projectName": "something"
    }
  }
}
```

#### Yeoman 4 .yo-rc.json:

`npm i -g yo@4.1.0 && yo ./generators/app`
``` json
{
  "generator-name": {
    "some config": "some value"
  }
}
```

#### Workaround Generator Code:

If I add `this.options.skipLocalCache = false;` to the generator, `yo@4.1.0` and `yo@3.1.1` both save the `promptValues`.

### Change

It looks like the root cause is that the default value for `skipLocalCache` changed to `true` even when not running in `experimental` mode [as part of this change](https://github.com/yeoman/environment/pull/244/files#diff-24984142d58a41889c987b60103bcd93653f3551f71f6cdf22b5762e885c9cd6R264-L268).

I don't see anything in the related commit, PR, or Release Notes that makes me think this was intentional.

**Comment by mshima** [comment]
This was intentional, but forgot the consequence for old generators.
Reverting will be another breaking change.

`store: true` is useful in some use cases where the answer won't change even using a different project.
Like author, preferred license, email.

Project specific should use:
```
        await super.prompt([{
            name: 'projectName',
            message: 'What is the name of your project?',
            [askAnswered: true] // to always show the prompt
        }], this.config);

        this.config.get('projectName');
```

**Comment by LTegtmeier** [comment]
I have existing projects that used `store: true` with Yeoman 3.x and I want to use those answers when running an updated generator.

If I use `store: true`, the generator will use the `promptValues` from the local `.yo-rc.json` but, will not update them if the user changes the existing value. 

To get the desired behavior, I can modify your solution to move the `promptValues` into `this.config` and clean them up so that there aren't duplicate values in the project `.yo-rc.json`:
``` js
const Generator = require('yeoman-generator');

module.exports = class extends Generator {
    async prompting() {
        this.config.set(this.config.get('promptValues'));
        await super.prompt([{
            name: 'projectName',
            message: 'What is the name of your project?',
            askAnswered: true
        }], this.config);
    }

    async writing() {
        const projectName = this.config.get('projectName');

        this.log(`The project name is: ${projectName}`)

        this.config.delete('promptValues');
    }
};
```

In general, I also want to save the answers in the `.yo-rc-global.json` because the answers are usually the same across projects but, may be custom for a specific project. If I add `store: true` to the above code, it will use the value in `.yo-rc-global.json` **over** the value in `this.config`. 

Basically, I want exactly the Yeoman 3 behavior to default to the local `.yo-rc.json` for existing projects, fall back to `.yo-rc-global.json` for new projects, and keep the values in sync in both. The only way I see to do that is to explicitly set `skipLocalCache = false`. Am I missing something? Is there a downside?

**Comment by mshima** [comment]
> I also want to save the answers in the `.yo-rc-global.json` because the answers are usually the same across projects but, may be custom for a specific project. If I add `store: true` to the above code, it will use the value in `.yo-rc-global.json` **over** the value in `this.config`.

Fixed in 5.3.0 https://github.com/yeoman/generator/commit/c1c847dbed005e40ded29ea46d1efd4dbac5cc51
